languages/dart/keywords/abstract.md

```diff
@@ -0,0 +1,47 @@
+# Abstract
+
+The keyword `abstract` is used to define an *abstract class* - a class that can not be instantiate directly, which means
+you can not create an object from it. Conversely, a class that can be instantiated directly is called a *concrete class*.
+
+Abstract types can define interfaces which hide the implementation details and showing only functionality to the user. 
+They let you focus on **what** the object does instead of **how** it does it.
+
+```dart
+// This class is declared abstract and thus can't be instantiated.
+abstract class CoffeeMachine {
+  //Body of abstract class
+}
+
+CoffeeMachine coffeeMachine = CoffeeMachine(); // Abstract classes can't be instantiated.
+```
+
+An abstract class can have constructors, fields, methods. Abstract methods define the class behaviour. They can only 
+exist within an abstract class and are delimiting by a semicolon (;).
+
+```dart
+abstract class CoffeeMachine {
+  void makeEspresso(); // Abstract method
+  void makeCoffee() { // Normal method
+    print("Making coffee");
+  }
+}
+```
```

Comment:
Add hint: Since abstract classes can have constructors and constructors in dart can omit their body. A constructor without a body does not denote that it is abstract.

Constructors in dart are always concrete.
```dart
class MyClass {
  // Constructor without a body
  MyClass.one();
  // Constructor with an empty body
  MyClass.two(){}
}
```